### PR TITLE
[RHACS] Updated upgrade process for RHACS 4.0

### PR DESCRIPTION
**Pull request by gaurav-nelson**
For https://issues.redhat.com/browse/ROX-12515

Cherry-pick into `rhacs-docs-4.0`

Preview: https://55453--docspreview.netlify.app/openshift-acs/latest/upgrading/upgrade-roxctl.html

### CODE REVIEW

**Comment by gaurav-nelson** [comment]
@c-du Can you please check these updates?

1. From the Google doc:
   > Follow the directions in the README file in the Central DB provisioning bundle to create Central DB and apply the configuration to Central

   Can you please provide that README instructions, I can add it in the `Creating resources by using the Central DB provisioning bundle` section.

2. Also do we neeed the Upgrading Scanner section?

File: modules/generate-central-database-provisioning-bundle.adoc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+----
+. Run the `central db generate` command:
++
+[source,terminal]
+----
+$ roxctl -n stackrox central db generate \\
+  <cluster_type> \\ <1>
+  <storage> \\ <2>
+  --output-dir <bundle_dir> \\ <3>
+  --central-db-image <central_db_image> <4>
```
Comment (gaurav-nelson) [inline modules/generate-central-database-provisioning-bundle.adoc:36]:
```suggestion
$ roxctl -n stackrox central db generate \
  <cluster_type> \ <1>
  <storage> \ <2>
  --output-dir <bundle_dir> \ <3>
  --central-db-image <central_db_image> <4>
```

**Comment by c-du** [comment]
An example of README contains the following instruction:
`  
- Deploy Central DB
     To deploy Postgres Central DB and prepare for Central upgrade:
     - Run scripts/setup.sh
     - Run ./deploy-central-db.sh
`


> @c-du Can you please check these updates?
> 
> 1. From the Google doc:
>    > Follow the directions in the README file in the Central DB provisioning bundle to create Central DB and apply the configuration to Central
>    
>    
>    Can you please provide that README instructions, I can add it in the `Creating resources by using the Central DB provisioning bundle` section.
> 2. Also do we neeed the Upgrading Scanner section?
Yes, we still need to Upgrading Scanner section but I do not expect changes in that secion.

**Review by c-du — COMMENTED** [review]
Gaurav,

We have decided not to GA in 74. Please hold the instruction to upgrade to Postgres until 3.75 aka 4.0. In 74, we still use the Tech review instruction as [here](https://docs.google.com/document/d/1JeckxriWV8VOJTYmgMigTAV5vH6ku-JtLwzj3ulC2hw/edit#), not finalized yet. I will update the jira request. Thanks.

File: modules/create-resource-central-db-bundle.adoc [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Module included in the following assemblies:
```
Comment (gaurav-nelson) [inline modules/create-resource-central-db-bundle.adoc:1]: Ignore this file, I'll delete this. Not required anymore.

File: upgrading/upgrade-roxctl.adoc [hunk 3]
```diff
@@ -1,35 +1,48 @@
...
-= Manually upgrading using the roxctl CLI
+= Manually upgrading using the `roxctl` CLI
 include::modules/common-attributes.adoc[]
 :context: upgrade-roxctl
 
 toc::[]
 
 [role="_abstract"]
-You can upgrade to the latest version of {product-title} from a supported older version.
+You can upgrade to the latest version of {product-title} from a supported older version. For upgrading to {product-title-short} 4.0, you must be using the latest patch release of {product-title-short} 3.74. If you are using an older version, you must first upgrade to {product-title-short} 3.74.
```
Comment (c-du) [inline upgrading/upgrade-roxctl.adoc:10]: Can we make this section stand out somehow? This step is critical and we do not want it to be missed.

**Review by c-du — COMMENTED** [review]
From our conversation, just post it for people's review. It looks like section “Patch the local role”, “Cleanup existing roles and role bindings”, “Remove the path” could be removed from now on since all upgrades are from 3.74.x and later.

File: modules/generate-central-database-provisioning-bundle.adoc [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+----
+. Run the `central db generate` command:
++
+[source,terminal,subs=attributes+]
+----
+$ roxctl -n stackrox central db generate \
+  <cluster_type> \ <1>
+  <storage> \ <2>
+  --output-dir <bundle_dir> \ <3>
+  --central-db-image registry.redhat.io/advanced-cluster-security/rhacs-central-db-rhel8:{rhacs-version} <4>
```
Comment (c-du) [inline modules/generate-central-database-provisioning-bundle.adoc:36]: The note 4 is not addressed in the following notes.

File: modules/create-resource-central-db-bundle.adoc (hunk 2 above)
Comment (c-du) [reply modules/create-resource-central-db-bundle.adoc:1]: Why did you plan to delete this? It looks useful to me. Any replacement for this?

**Review by c-du — COMMENTED** [review]
Hi, Gaurav, can you update the preview link also? It seems the preview does not have the most recent update. And also, do you have updates to the helm chart upgrade somewhere?

File: modules/create-resource-central-db-bundle.adoc (hunk 2 above)
Comment (gaurav-nelson) [reply modules/create-resource-central-db-bundle.adoc:1]: Sorry, I commented on the wrong PR on a wrong file.

**Review by c-du — COMMENTED** [review]
There are a few things we can remove now since we have requested customer to upgrade to 3.74 first.
1. In Rollbacking back Central -> Rolling back Central normally -> Prerequisites, remove "You must be using Red Hat Advanced Cluster Security for Kubernetes 3.0.57.0 or higher.". In the second item, remove or change the warning "you will not be able to roll back to an earlier version." to something like "you may not be able to ...". Because in 4.0, we have special arrangement to provide the normal rollback. 
2. In Rollbacking back Central -> Rolling back Central forcefully -> Prerequisites, remove "You must be using Red Hat Advanced Cluster Security for Kubernetes 3.0.58.0 or higher."

**Pull request merged**